mobile/library/cc/engine_builder.cc

```diff
@@ -18,11 +38,39 @@ void insertCustomFilter(const std::string& filter_config, std::string& config_te
   absl::StrReplaceAll({{"#{custom_filters}", absl::StrCat("#{custom_filters}\n", filter_config)}},
                       &config_template);
 }
+
+bool generatedStringMatchesGeneratedBoostrap(
+    std::string& config_str, std::unique_ptr<envoy::config::bootstrap::v3::Bootstrap> bootstrap) {
+  Thread::SkipAsserts skip;
+  ProtobufMessage::StrictValidationVisitorImpl visitor;
+  envoy::config::bootstrap::v3::Bootstrap config_bootstrap;
+  MessageUtil::loadFromYaml(absl::StrCat(config_header, config_str), config_bootstrap, visitor);
+
+  Protobuf::util::MessageDifferencer differencer;
+  differencer.set_message_field_comparison(Protobuf::util::MessageDifferencer::EQUIVALENT);
+  differencer.set_repeated_field_comparison(Protobuf::util::MessageDifferencer::AS_SET);
+
+  bool same = differencer.Compare(config_bootstrap, *bootstrap);
+
+  if (!same) {
+    std::cerr << config_bootstrap.DebugString();
+    std::cerr << "============================================";
+    std::cerr << bootstrap->DebugString();
+  }
+  return same;
+}
+
 } // namespace
 
 EngineBuilder::EngineBuilder(std::string config_template)
-    : callbacks_(std::make_shared<EngineCallbacks>()), config_template_(config_template) {}
-EngineBuilder::EngineBuilder() : EngineBuilder(std::string(config_template)) {}
+    : callbacks_(std::make_shared<EngineCallbacks>()), config_template_(config_template) {
+  config_bootstrap_incompatible_ = true;
```

Comment:
nit: This should probably go into the initializer list. (Same below)

Reply:
moved this one, prefer keeping the other where it is  - we're using the base constructor then overriding one variable.